### ISSUE TITLE
A way to pass RTS options to the GHC

### ISSUE THREAD

**Issue by phadej** [commercialhaskell/stack#3353]
Previously I have using
```
stack build ... --ghc-options='+RTS -M4G -RTS'
```

but https://github.com/commercialhaskell/stack/pull/3316 *fixed* the loophole, now stack generates
```
--ghc-options +RTS --ghc-options -M4G --ghc-options -RTS
```

**Comment by mgsloan** [comment]
@phadej Thanks for the report!  Marking this as blocking release, as I'd rather not release a change to ghc-options only to revert it.  One way to handle this is to add special parsing of RTS options.

Actually, I think that this was never a problem for configure-options, just an issue when invoking ghci directly with multi-ghc-options.  I noticed that the configuration parsing code for stack.yaml uses argument parsing for "ghc-options", when it probably shouldn't.  So, steps would be:

1. Use parseArgs for options passed to ghci
2. Remove use of parseArgs for `--ghc-options` in cli and `ghc-options: ` in config

**Comment by mgsloan** [comment]
In preparation for the release, I've reverted the fix to #3316.  Thanks a bunch for bringing this to my attention!

Hopefully by next release #3316 will also be fixed without breaking this usecase.
